apt-hook/hook.cc

```diff
@@ -24,19 +24,42 @@
 #include <apt-pkg/strutl.h>
 
 #include <fstream>
+#include <iostream>
 #include <sstream>
 #include <string>
+#include <vector>
 
 #include <assert.h>
 #include <sys/stat.h>
 #include <libintl.h>
 #include <locale.h>
 
+#define CONTRACT_EXPIRY_STATUS_MESSAGE_TEMPLATE_PATH              "/var/lib/ubuntu-advantage/messages/contract-expiry-status.tmpl"
+#define CONTRACT_EXPIRY_STATUS_MESSAGE_STATIC_PATH                "/var/lib/ubuntu-advantage/messages/contract-expiry-status"
+#define CONTRACT_EXPIRED_APT_UPGRADE_MESSAGE_TEMPLATE_PATH        "/var/lib/ubuntu-advantage/messages/contract-expired-apt-upgrade.tmpl"
+#define CONTRACT_EXPIRED_APT_UPGRADE_MESSAGE_STATIC_PATH          "/var/lib/ubuntu-advantage/messages/contract-expired-apt-upgrade"
+#define CONTRACT_EXPIRED_APT_DIST_UPGRADE_MESSAGE_TEMPLATE_PATH   "/var/lib/ubuntu-advantage/messages/contract-expired-apt-dist-upgrade.tmpl"
+#define CONTRACT_EXPIRED_APT_DIST_UPGRADE_MESSAGE_STATIC_PATH     "/var/lib/ubuntu-advantage/messages/contract-expired-apt-dist-upgrade"
```

Comment:
@orndorffgrant /var/lib/ubuntu-advantage/messages/contract-expiry-status.tmpl will contain the following
```bash
CAUTION: Your UA Apps: ESM service will expire in 10 days.        
Renew UA subscription at https://ubuntu.com/advantage to ensure                                        
continued security coverage for your applications.
```

What messages to you forsee as content in /var/lib/ubuntu-advantage/messages/contract-expired-apt-upgrade.tmpl and /var/lib/ubuntu-advantage/messages/contract-expired-apt-dist-upgrade.tmpl that differs?

Reply:
"/var/lib/ubuntu-advantage/messages/contract-expiry-status.tmpl" -> https://docs.google.com/document/d/1bNvQ2EveGdGyMqs9WxoLEt1U-x9nqEbUBtdNzIpGAXE/edit#heading=h.aj8kpel9nomu

These next two aren't necessary if we don't actually need different messages per upgrade/dist-upgrade
"/var/lib/ubuntu-advantage/messages/contract-expired-apt-upgrade.tmpl" -> see apt upgrade message here https://docs.google.com/document/d/1bNvQ2EveGdGyMqs9WxoLEt1U-x9nqEbUBtdNzIpGAXE/edit#heading=h.fv9w5coai2c1
"/var/lib/ubuntu-advantage/messages/contract-expired-apt-dist-upgrade.tmpl" -> see apt dist-upgrade message here https://docs.google.com/document/d/1bNvQ2EveGdGyMqs9WxoLEt1U-x9nqEbUBtdNzIpGAXE/edit#heading=h.fv9w5coai2c1

Not sure if these next two are necessary. We might be able to use just one banner message here. If we keep them separate though, it gives us the opportunity to only print if esm_apps_count > 0 for example (not currently implemented)
"/var/lib/ubuntu-advantage/messages/esm-apps-not-enabled.tmpl" -> apps version of https://docs.google.com/document/d/1bNvQ2EveGdGyMqs9WxoLEt1U-x9nqEbUBtdNzIpGAXE/edit#heading=h.jcfd4jpjjhqv
"/var/lib/ubuntu-advantage/messages/esm-infra-not-enabled.tmpl" -> infra version of https://docs.google.com/document/d/1bNvQ2EveGdGyMqs9WxoLEt1U-x9nqEbUBtdNzIpGAXE/edit#heading=h.jcfd4jpjjhqv